pkg/cloudprovider/providers/openstack/openstack.go

```diff
@@ -886,43 +932,30 @@ func (os *OpenStack) getVolume(diskName string) (volumes.Volume, error) {
 	return volume, err
 }
 
-func (os *OpenStack) getComputeIDbyHostname(cClient *gophercloud.ServiceClient) (string, error) {
+// getLocalComputeID tries to find ID of the OpenStack instance where
+// this code runs.
+func (os *OpenStack) getLocalComputeID(cClient *gophercloud.ServiceClient) (string, error) {
+	// try to read /var/lib/cloud/data/instance-id
+	const instanceIDFile = "/var/lib/cloud/data/instance-id"
+	idBytes, err := ioutil.ReadFile(instanceIDFile)
+	if err == nil {
+		id := string(idBytes)
+		id = strings.TrimSpace(id)
+		glog.V(5).Infof("Got instance id from %s: %s", instanceIDFile, id)
+		return id, nil
+	}
+	glog.V(4).Infof("Reading %s failed: %v, trying hostname", instanceIDFile, err)
 
+	// try to find ID from hostname
```

Comment:
> Regardless of the method used to fetch it, we should cache value in memory since it will never change and will be used multiple times by the other functions in this file.

Ok, patch will follow shortly.

> Just fetch http://169.254.169.254/openstack/2012-08-10/meta_data.json

`http://169.254.169.254/2007-01-19/meta-data/instance-id` looks easier to parse. Can it be used instead of the JSON? Is it stable enough?